hera_mc/cm_hookup.py

```diff
@@ -398,24 +399,41 @@ def __make_header_row(self, header_col_list, cols_to_show):
                 headers.append(col)
         return headers
 
-    def __make_table_row(self, hup_list, headers, timing, show_level):
+    def __make_table_row(self, hup_list, headers, timing, show_level, show_port, show_rev):
         td = ['-'] * len(headers)
         dip = ''
         j = 0  # This catches potentially duplicated part_type names
         for d in hup_list:
             part_type = self.part_type_cache[d.upstream_part]
             if part_type == headers[0]:
-                td[0] = d.upstream_part + ':' + d.up_part_rev + ' <' + d.upstream_output_port
+                td[0] = d.upstream_part
+                if show_rev:
+                    td[0] += ':' + d.up_part_rev
+                if show_port:
+                    td[0] += ' <' + d.upstream_output_port
                 dip = d.downstream_input_port + '> '
                 j += 1
-            else:
-                if part_type in headers:
-                    td[headers.index(part_type, j)] = dip + d.upstream_part + ':' + d.up_part_rev + ' <' + d.upstream_output_port
-                    j += 1
+            elif part_type in headers:
+                if show_port:
+                    td[headers.index(part_type, j)] = dip
+                else:
+                    td[headers.index(part_type, j)] = ''
+                td[headers.index(part_type, j)] += d.upstream_part
+                if show_rev:
+                    td[headers.index(part_type, j)] += ':' + d.up_part_rev
+                if show_port:
+                    td[headers.index(part_type, j)] += ' <' + d.upstream_output_port
```

Comment:
It would be a bit more readable, I think, to construct this string as a local variable, then assign it to `td[headers.index(part_type, j)]` after all of the conditionals have been worked through.